engine/apps/schedules/tasks/shift_swaps/slack_followups.py

```diff
@@ -0,0 +1,92 @@
+import datetime
+
+from celery.utils.log import get_task_logger
+from django.conf import settings
+from django.core.cache import cache
+from django.utils import timezone
+
+from apps.schedules.models import ShiftSwapRequest
+from apps.slack.scenarios.shift_swap_requests import ShiftSwapRequestFollowUp
+from common.custom_celery_tasks import shared_dedicated_queue_retry_task
+
+task_logger = get_task_logger(__name__)
+
+# When to send followups before the swap start time
+FOLLOWUP_OFFSETS = [
+    datetime.timedelta(weeks=4),
+    datetime.timedelta(weeks=3),
+    datetime.timedelta(weeks=2),
+    datetime.timedelta(weeks=1),
+    datetime.timedelta(days=3),
+    datetime.timedelta(days=2),
+    datetime.timedelta(days=1),
+    datetime.timedelta(hours=12),
+]
+
+# FOLLOWUP_WINDOW is used by _get_shift_swap_requests_in_followup_window and _mark_followup_sent to:
+# 1. Determine which SSRs to send followups for when the periodic task is run
+# 2. Prevent sending multiple followups for a single SSRS in a short period
+FOLLOWUP_WINDOW = datetime.timedelta(hours=1)
+
+
+@shared_dedicated_queue_retry_task()
+def send_shift_swap_request_slack_followups() -> None:
+    """A periodic task to send Slack followups for shift swap requests."""
+
+    for shift_swap_request in _get_shift_swap_requests_in_followup_window(timezone.now()):
+        if not _has_followup_been_sent(shift_swap_request):
+            send_shift_swap_request_slack_followup.delay(shift_swap_request.pk)
+            _mark_followup_sent(shift_swap_request)
+
+
+def _get_shift_swap_requests_in_followup_window(now: datetime.datetime) -> list[ShiftSwapRequest]:
+    """Get all SSRs that are in the followup window."""
+
+    shift_swap_requests_in_notification_window = []
+    for shift_swap_request in ShiftSwapRequest.objects.filter(benefactor__isnull=True, swap_start__gt=now):
```

Comment:
_nit_ on the query. We already have this same query [here](https://github.com/grafana/oncall/blob/dev/engine/apps/mobile_app/tasks.py#L514). wdyt about moving it as a method on the `QuerySet` (or `Manager`, not sure which one is more appropriate)? Something like, `get_open_requests`.

This way if we need to change the definition of this query, it only needs to be done once.

Reply:
Good call, addressed in https://github.com/grafana/oncall/pull/2798/commits/7fe839dd39ceb9706f52ecf1cda17257fbf14555